src/main/java/org/embulk/input/jira/JiraInputPlugin.java

```diff
@@ -144,34 +159,38 @@ public TaskReport run(final TaskSource taskSource,
             }
             pageBuilder.finish();
         }
-        return Exec.newTaskReport();
+        return CONFIG_MAPPER_FACTORY.newTaskReport();
     }
 
     @Override
     public ConfigDiff guess(final ConfigSource config)
     {
         // Reset columns in case already have or missing on configuration
         config.set("columns", new ObjectMapper().createArrayNode());
-        final PluginTask task = config.loadConfig(PluginTask.class);
+        final PluginTask task = CONFIG_MAPPER.map(config, PluginTask.class);
         JiraUtil.validateTaskConfig(task);
         final JiraClient jiraClient = getJiraClient();
         jiraClient.checkUserCredentials(task);
         final List<Issue> issues = jiraClient.searchIssues(task, 0, GUESS_RECORDS_COUNT);
         if (issues.isEmpty()) {
             throw new ConfigException("Could not guess schema due to empty data set");
         }
+
+        // TODO: Eliminate the use of ExecInternal.getInjector that soon will be obsolete.
         final Buffer sample = Buffer.copyOf(createSamples(issues, getUniqueAttributes(issues)).toString().getBytes());
-        final JsonNode columns = Exec.getInjector().getInstance(GuessExecutor.class)
-                                .guessParserConfig(sample, Exec.newConfigSource(), createGuessConfig())
-                                .get(JsonNode.class, "columns");
-        return Exec.newConfigDiff().set("columns", columns);
+        final JsonNode columns = ExecInternal.getInjector().getInstance(GuessExecutor.class)
+                .guessParserConfig(sample, Exec.newConfigSource(), createGuessConfig())
+                .get(JsonNode.class, "columns");
+
+        return CONFIG_MAPPER_FACTORY.newConfigDiff().set("columns", columns);
     }
 
     private ConfigSource createGuessConfig()
     {
+        // TODO: there will be no need for this after "Eliminate the use of ExecInternal.getInjector that soon will be obsolete."
         return Exec.newConfigSource()
-                    .set("guess_plugins", ImmutableList.of("jira"))
-                    .set("guess_sample_buffer_bytes", GUESS_BUFFER_SIZE);
+                .set("guess_plugins", ImmutableList.of("jira"))
```

Comment:
Should `Exec.newConfigSource()` changes to `CONFIG_MAPPER_FACTORY.newConfigSource()` ?

Reply:
(similar to the above) It has to be `Exec.newConfigSource` since the ConfigSource will be consumed by GuessExecutor, which only works with the "old-style" ones.

Reply:
Ok got it